jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/milestones/CaseMilestoneListPresenter.java

```diff
@@ -48,19 +59,56 @@ protected void clearCaseInstance() {
 
     @Override
     protected void loadCaseInstance(final CaseInstanceSummary cis) {
-        refreshData(caseId);
+        refreshData();
+    }
+    
+    @Override
+    public void setPageSize() {
+        this.pageSize = PAGE_SIZE;          
     }
 
     protected void searchCaseMilestones() {
-        refreshData(caseId);
+        refreshData();
+    }
+
+    protected void refreshData() {
+        milestonesServiceCall(caseId,
+                              getCurrentPage());
+    }
+
+    protected void milestonesServiceCall(String caseId,
+                                         int currentPage) {
+        caseService.call((List<CaseMilestoneSummary> milestones) -> {
+            visibleCaseMilestones.addAll(milestones);
+            view.setCaseMilestoneList(visibleCaseMilestones.stream()
+                                          .collect(toList()));
+        }).getCaseMilestones(containerId,
+                             caseId,
+                             view.getCaseMilestoneSearchRequest(),
+                             getCurrentPage(),
+                             getPageSize());
+
+        loadButtonToggle();
     }
 
-    protected void refreshData(String caseId) {
+    protected void loadMoreCaseMilestones() {
+        setCurrentPage(getCurrentPage() + 1);
+        milestonesServiceCall(caseId,
+                              getCurrentPage());
+    }
+
+    private void loadButtonToggle() {
```

Comment:
Instead of loading the next page of items, the load button should be visible if loading the page returned the same number of items as the page size. This way we can avoid this extra call and keep consistency with other search UI behavior.